packages/next/src/build/create-compiler-aliases.ts

```diff
@@ -220,12 +220,16 @@ export function createNextApiEsmAliases() {
   return aliasMap
 }
 
-export function createAppRouterApiAliases() {
-  const mapping = {
+export function createAppRouterApiAliases(isServerOnlyLayer: boolean) {
+  const mapping: Record<string, string> = {
     head: 'next/dist/client/components/noop-head',
     dynamic: 'next/dist/api/app-dynamic',
   }
 
+  if (isServerOnlyLayer) {
+    mapping['navigation'] = 'next/dist/api/navigation.react-server'
```

Comment:
should we have same for the turbopack? or it just passes new test automatically?

Reply:
@kwonoj good call, we can add for turbopack too

Reply:
Tested it that turbopack doesn't have this issue, I'll create a separate PR to follow up the alias for turbopack